src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -574,6 +586,20 @@ internal int ExecuteTool(string pathToTool, string responseFileCommands, string
             }
 
             return ExitCode;
+
+            // Construct the friendly name for the compilation. This does not need to be unique. Instead
+            // it's used by developers to understand what compilation is running on the server.
```

Comment:
Can it happen in reality that developers want to disambiguate requests? Maybe the friendly name could also include a GUID for that.

Also someone could be using the Csc task without setting ProjectName and TargetFramework, I imagine, and debugging the logs would be easier if there was a fallback GUID.

Reply:
Yep. Will add the `Guid` in the fallback case.